### PR TITLE
Refactor Codes into own service

### PR DESCRIPTION
**Pull request by dhaselhan**
* Reduce duplicated code services into dedicated code service
* Make a single http call to get all codes

### CODE REVIEW

File: alcs-api/src/application/application-code/application-code.service.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @InjectRepository(ApplicationDecisionMaker)
+    private decisionMakerRepository: Repository<ApplicationDecisionMaker>,
+  ) {}
+
+  async getAllCodes() {
+    const values = await Promise.all([
+      this.typeRepository.find(),
+      this.statusRepository.find(),
+      this.decisionMakerRepository.find(),
+    ]);
```
Comment (pdittaro) [inline alcs-api/src/application/application-code/application-code.service.ts:24]: So two part mapping to get codes? One to stick it in the service and one to explicitly expose it on the controller?

Comment (dhaselhan) [reply alcs-api/src/application/application-code/application-code.service.ts:24]: Yes. Services should be the ones interacting with the DB and controllers handling mapping to the front end.

Comment (dhaselhan) [reply alcs-api/src/application/application-code/application-code.service.ts:24]: I could probably cache the ones that are static, e.g. everything but status, and load them once when the service starts.

**Review by mhuseinov — APPROVED** [review]
👍

**Pull request merged**